src/toil/wdl/wdl_analysis.py

```diff
@@ -776,16 +828,21 @@ def parse_declaration_expressn_memberaccess(self, lhsAST, rhsAST, es):
         if isinstance(lhsAST, wdl_parser.Terminal):
             es = es + lhsAST.source_string
         elif isinstance(lhsAST, wdl_parser.Ast):
-            raise NotImplementedError
+            es = es + self.parse_declaration_expressn(lhsAST, es)
         elif isinstance(lhsAST, wdl_parser.AstList):
             raise NotImplementedError
 
-        es = es + '_'
+        # hack-y way to make sure pair.left and pair.right are parsed correctly.
```

Comment:
Can you explain a bit more about why this hack is like it is? What's the underscore separator doing in the first place? Have we elsewhere turned all the hierarchical return types into flat underscore-separated keys?

Reply:
The underscore separator is for task outputs, I believe.

```wdl
workflow {
  call t
  call t2 { input: in=t.out }
}
```
For this workflow, `t.out` is changed to `t_out = t.rv("out")`. See: https://github.com/DataBiosphere/toil/blob/releases/4.2.x/src/toil/wdl/wdl_synthesis.py#L376

In this case, we want to access `left` and `right` inside WDLPair. 

This can potentially cause problems if someone has an output variable named left or right, but this is the best way I could think of without passing the type into `parse_declaration_expressn()` everywhere.

Reply:
I think this needs to be rethought overall; any acceptable WDL variable name needs to be allowed in user code, not just the ones that don't upset our parser. But it looks like it might be a bigger issue than just this ticket, and need some kind of unified `escape_user_variable_name()` function.